src/Microsoft.Data/ArrowStringColumn.cs

```diff
@@ -0,0 +1,378 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+using Apache.Arrow;
+using Apache.Arrow.Types;
+
+namespace Microsoft.Data
+{
+    /// <summary>
+    /// An immutable to hold Arrow style strings
+    /// </summary>
+    public partial class ArrowStringColumn : BaseColumn
+    {
+        private IList<ReadOnlyDataFrameBuffer<byte>> _dataBuffers;
+        private IList<ReadOnlyDataFrameBuffer<int>> _offsetsBuffers;
+        private IList<ReadOnlyDataFrameBuffer<byte>> _nullBitMapBuffers;
+
+        public ArrowStringColumn(string name) : base(name, 0, typeof(string))
+        {
+            _dataBuffers = new List<ReadOnlyDataFrameBuffer<byte>>();
+            _offsetsBuffers = new List<ReadOnlyDataFrameBuffer<int>>();
+            _nullBitMapBuffers = new List<ReadOnlyDataFrameBuffer<byte>>();
+        }
+
+        public ArrowStringColumn(string name, ReadOnlyDataFrameBuffer<byte> dataBuffer, ReadOnlyDataFrameBuffer<int> offsetBuffer, ReadOnlyDataFrameBuffer<byte> nullBitMapBuffer, int length, int nullCount) : base(name, length, typeof(string))
```

Comment:
I'll remove this once Arrow's `StringColumn.Builder.Append` supports appending null values. Leaving this around for the unit tests now